.ci/check_python_dists.sh

```diff
@@ -25,7 +25,7 @@ if [ $PY_MINOR_VER -gt 7 ]; then
         pydistcheck \
             --inspect \
             --ignore 'compiled-objects-have-debug-symbols,distro-too-large-compressed' \
-            --max-allowed-size-uncompressed '60M' \
+            --max-allowed-size-uncompressed '70M' \
```

Comment:
With generating code for 2 more CUDA architectures, the linux CUDA wheels are now around 63 MB

```text
==================== running pydistcheck ====================

checking '/LightGBM/dist/lightgbm-4.1.0.99-py3-none-manylinux_2_27_x86_64.whl'
----- package inspection summary -----
file size
  * compressed size: 40.6M
  * uncompressed size: 63.3M
  * compression space saving: 35.9%
contents
  * directories: 0
  * files: 17 (1 compiled)
size by extension
  * .so - 64423.5K (99.3%)
  * .py - 404.7K (0.6%)
  * no-extension - 21.5K (0.0%)
  * .txt - 0.0K (0.0%)
  * .typed - 0.0K (0.0%)
```

Build link: https://github.com/microsoft/LightGBM/actions/runs/6186545556/job/16794443363?pr=6099

We could start dropping older architectures at some point if we think that's a problem, but I don't think that should even be considered until we start possibly distributing precompiled CUDA wheels via a repository like PyPI.

Until then, I think building for more architectures to support a wider range of GPUs is desirable, and a few more MB on disk for the wheel isn't a big concern.